---
title: "Approach & Outputs"
---

We exploit new data types and new types of experiments and studies by developing the computational techniques needed to turn raw data into biology.

- **Multi-scale biology in space and time**: bringing together different data types and resolutions to find low-dimensional explanations (factors, gradients, regions, graphs and networks) of high-dimensional data, using statistical models, first-principles based theory and machine learning.
- **Spatial omics in immunooncology** to find and improve treatment options for patients.
- **Multidimensional phenotypes from genetic and drug-based perturbation assays** to map context-dependent gene-gene and gene-drug interaction networks.
- Many powerful mathematical and computational ideas exist but are difficult to access. We aim to translate them into practical methods and software that make a real difference to biomedical researchers. We sometimes term this approach **translational statistics**.


::: {.column-margin}
![Modern Statistics for Modern Biology textbook, with Susan Holmes: [online version](https://www.huber.embl.de/msmb/). There is also a [print version published by CUP](https://www.cambridge.org/it/universitypress/subjects/statistics-probability/statistics-life-sciences-medicine-and-health/modern-statistics-modern-biology?format=HB).](photos/MSFMB-Cover2-compressed.jpg){width=240 fig-align="center"}

![Cellular neighborhood analysis of healthy and malignant lymph nodes based on single-cell resolution spatial proteomics by multiplexed immunohistochemistry.](photos/cellularneighborhoods.png){width=240 fig-align="center"}

![Cluster-free differential expression analysis of sc-RNA-seq data using LEMUR. [Paper link](https://doi.org/10.1038/s41592-023-01814-1).](photos/clusterfreeDE-2048x1649.png){width=240 fig-align="center"}

![Comparison of transformations for single-cell RNA-seq data. [Paper link](https://doi.org/10.1038/s41592-023-01814-1).](photos/Transformations-Fig1c.png){width=240 fig-align="center"}

![Ternary plots of relative sensitivities to targeted kinase inhibitors for a cohort of primary tumour samples of chronic lymphocytic leukaemia (CLL). [Paper link](https://doi.org/10.1172/JCI93801).](photos/JCI2018-fig_02_l.jpg){width=240 fig-align="center"}
:::

### Spatial omics and imaging

Modern spatial omics techniques measure the spatial distribution of tens of thousands distinct molecules, with major limiting factors being sampling efficiency and spatial resolution. Photon microscopy observes one or a few distinct molecules at resolutions of tens of nanometers. Electron microscopy measures e$^-$ densities at resolutions of a few angstroms. But how to bring this all together? How to navigate huge Terabyte-scale maps of cells and organisms? How to decipher biological functions and processes, associate them with phenotypes in health and disease, and exploit this for better understanding fundamental biology and advance biotechnology and biomedicine?

### Functional precision medicine and immuno-oncology

We integrate observational 'omics data, interventional clinical data, and systematic genetic or chemical perturbation data on (ex-vivo) model systems to decipher the molecular mechanisms of variable sensitivity and resistance of tumors to treatments (precision oncology collaboration with Thorsten Zenz at University Hospital Zurich), and to understand the role of the immune system and the tumour microenvironment in tumourigenesis, progression and treatment (systems immunology collaboration with Sascha Dietrich at University Hospital Düsseldorf).

### Open science

As we engage with new data types, we aim to develop high-quality computational methods of wide applicability. We consider the release and maintenance of scientific software an integral part of doing science. We contribute to the [Bioconductor] project, an open source software collaboration to provide tools for the analysis and understanding of genome-scale data. An [example is our DESeq2 package](posts/muchcitedpapers.html) for analyzing count data from high-throughput sequencing.

### Mentoring and career development

Science is an intellectual adventure and a creative process done by people. For each of us, our work is at the same time, a means to achieve a scientific goal, a job that enables us pay our bills, and a stage of training and professional development. This includes student internships, BSc/MSc theses, PhD theses, postdoctoral projects. The group, and EMBL more generally, offers a well-established mentoring framework to support these triple objectives. [Former group members](people.html#alumni) have moved on to rewarding careers: professors, independent group leaders, senior management or professional scientist roles in industry.

### Teaching

We maintain the textbook Modern Statistics for Modern Biology by Susan Holmes and Wolfgang Huber. The book is available [online, for free, as HTML](https://www.huber.embl.de/msmb/). It was published as a [printed book in 2019 by Cambridge University Press](https://www.cambridge.org/it/universitypress/subjects/statistics-probability/statistics-life-sciences-medicine-and-health/modern-statistics-modern-biology?format=HB).

We run the annual summer school CSAMA—Biological Data Science. It usually takes place in June in Brixen/Bressanone. [Here is the webpage of the 2025 edition](https://csama2025.bioconductor.eu). See [here for some impressions](https://www.huber.embl.de/group/posts/2023-06-16-csama.html).

In July 2023, 2024 and 2025, we co-organized the [Ukrainian Biological Data Science Summer School](https://www.bds3.org) in Uzhhorod, Ukraine. See also [Wolfgang's post about it](posts/2025-07-ubds3.html).

We develop publicly available interactive [training materials on statistical methods](https://www.ebi.ac.uk/training/online/courses/biostatistics-introduction/). 

### Software

We are a frequent contributor to the Bioconductor project

|              |                                    |
|--------------|------------------------------------|
|[LEMUR](https://github.com/const-ae/lemur/)|Cluster-free differential expression analysis of multi-condition single-cell data using Latent Embedding Multivariate Regression|
|[MOFA](https://biofam.github.io/MOFA2/)|Multi-Omics Factor Analysis|
|[DESeq2](https://bioconductor.org/packages/DESeq2)|Differential gene expression analysis based on the negative binomial distribution|
|[IHW](https://bioconductor.org/packages/IHW)|Multiple testing and false discovery rate (FDR) control by Independent Hypothesis Weighting|
|[EBImage](https://bioconductor.org/packages/EBImage)|Image processing and analysis toolbox for R|
|[Rarr](https://bioconductor.org/packages/Rarr)|Read Zarr Files in R|
|[rhdf5](https://bioconductor.org/packages/rhdf5)|R Interface to HDF5|
|[vsn](https://bioconductor.org/packages/vsn)|Normalization and variance stabilizing transformation of fluorescence intensity data|
|[cellHTS2](https://bioconductor.org/packages/cellHTS2)|Analysis of cell-based high-throughput screens|
|[DEXSeq](https://bioconductor.org/packages/DEXSeq)|Inference of differential exon usage in RNA-Seq|
|[HilbertVis](https://bioconductor.org/packages/HilbertVis)|Visualize long vectors of data using Hilbert curves|
| | |
|**Python**| |
|[HTSeq](https://htseq.readthedocs.io/en/latest/)|Processing and analyzing data from high-throughput sequencing assays|
|[SOFA](https://github.com/tcapraz/SOFA/)|Semi-supervised (Multi) Omics Factor Analysis|
|[spatialproteomics](https://github.com/sagar87/spatialproteomics/)|lightweight wrapper around xarray to facilitate processing, exploration and analysis of multiplexed immunohistochemistry data|


: {tbl-colwidths="[24,76]"}

{{< include _links.qmd >}}

